services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/RecordHandler.java

```diff
@@ -756,6 +765,75 @@ private void doConfigurePort(final CommandMessage message, final String replyToT
         }
     }
 
+    private void doDumpSwitchPortsDescriptionRequest(
+            CommandMessage message, String replyToTopic, Destination replyDestination) {
+        DumpSwitchPortsDescriptionRequest request = (DumpSwitchPortsDescriptionRequest) message.getData();
+        try {
+            SwitchId switchId = request.getSwitchId();
+            logger.debug("Get port descriptions for switch {}", switchId);
+
+            SwitchPortsDescription response = getSwitchPortsDescription(switchId);
+
+            InfoMessage infoMessage = new InfoMessage(response, message.getTimestamp(), message.getCorrelationId());
+            context.getKafkaProducer().postMessage(replyToTopic, infoMessage);
+        } catch (SwitchOperationException e) {
+            logger.error("Unable to dump switch port descriptions request", e);
+            ErrorData errorData =
+                    new ErrorData(
+                            ErrorType.NOT_FOUND, e.getMessage(), "Unable to dump switch port descriptions request");
+            ErrorMessage error =
+                    new ErrorMessage(
+                            errorData, System.currentTimeMillis(), message.getCorrelationId(), replyDestination);
+            context.getKafkaProducer().postMessage(replyToTopic, error);
+        }
+    }
+
+    private SwitchPortsDescription getSwitchPortsDescription(SwitchId switchId) throws SwitchOperationException {
+
+        List<OFPortDesc> ofPortsDescriptions =
+                context.getSwitchManager().dumpPortsDescription(DatapathId.of(switchId.toLong()));
+        List<PortDescription> portsDescriptions = ofPortsDescriptions.stream()
+                .map(OfPortDescConverter::toPortDescription)
+                .collect(Collectors.toList());
+
+        return SwitchPortsDescription.builder()
+                .version(ofPortsDescriptions.get(0).getVersion().toString())
+                .portsDescription(portsDescriptions)
+                .build();
+    }
+
+    private void doDumpPortDescriptionRequest(
+            CommandMessage message, String replyToTopic, Destination replyDestination) {
+        DumpPortDescriptionRequest request = (DumpPortDescriptionRequest) message.getData();
+        try {
+            SwitchId switchId = request.getSwitchId();
+            logger.debug("Get port description for switch {}", switchId);
+            SwitchPortsDescription switchPortsDescription = getSwitchPortsDescription(switchId);
+
+            int port = request.getPortNumber();
+            PortDescription response =
+                    switchPortsDescription
+                            .getPortsDescription()
+                            .stream()
```

Comment:
No reason to split "PortDescription response = switchPortsDescription.getPortsDescription()" on 3 lines. This just negatively affects readability.